liquidity/loopout_builder.go

```diff
@@ -0,0 +1,160 @@
+package liquidity
+
+import (
+	"context"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/lightninglabs/loop"
+	"github.com/lightninglabs/loop/labels"
+	"github.com/lightninglabs/loop/loopdb"
+	"github.com/lightninglabs/loop/swap"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// Compile-time assertion that loopOutBuilder satisfies the swapBuilder
+// interface.
+var _ swapBuilder = (*loopOutBuilder)(nil)
+
+func newLoopOutBuilder(cfg *Config) *loopOutBuilder {
+	return &loopOutBuilder{
+		cfg: cfg,
+	}
+}
+
+type loopOutBuilder struct {
+	// cfg contains all the external functionality we require to create
+	// swaps.
+	cfg *Config
+}
+
+// swapType returns the swap type that the builder is responsible for creating.
+func (b *loopOutBuilder) swapType() swap.Type {
+	return swap.TypeOut
+}
+
+// maySwap checks whether we can currently execute a swap, examining the
+// current on-chain fee conditions against relevant to our swap type against
+// our fee restrictions.
+//
+// For loop out, we check whether the fees required for our on-chain sweep
+// transaction exceed our fee limits.
+func (b *loopOutBuilder) maySwap(ctx context.Context, params Parameters) error {
+	estimate, err := b.cfg.Lnd.WalletKit.EstimateFee(
```

Comment:
I think it's ok in this case, we can always calculate the fee with known tx weight. Here we're looking at a parameter which says "I only want to autoloop if it's going to cost me less than x sat/byte". Easier way to frame it to the user since they may not have a good idea of the size of the swap tx.

Reply:
It just returns a sats/KWeight value that can later on be used with the actual tx.

Reply:
The end user actually specifies a fee rate limit here in sat/byte so this is actually what we want.